dotnet/src/SemanticKernel/AI/OpenAI/Clients/OpenAIClientAbstract.cs

```diff
@@ -112,6 +155,58 @@ protected async Task<IList<Embedding<float>>> ExecuteEmbeddingRequestAsync(strin
         }
     }
 
+    /// <summary>
+    /// Run the HTTP request to generate a list of images
+    /// </summary>
+    /// <param name="url">URL for the chat request API</param>
+    /// <param name="requestBody">Request payload</param>
+    /// <param name="cancellationToken">Cancellation token</param>
+    /// <returns>List of image URLs</returns>
+    /// <exception cref="AIException">AIException thrown during the request.</exception>
+    protected async Task<IList<string>> ExecuteImageUrlGenerationRequestAsync(
+        string url,
+        string requestBody,
+        CancellationToken cancellationToken = default)
+    {
+        try
+        {
+            var result = await this.ExecutePostRequestAsync<ImageUrlGenerationResponse>(url, requestBody, cancellationToken);
+            return result.Images.Select(x => x.Url).ToList();
+        }
+        catch (Exception e) when (e is not AIException)
+        {
+            throw new AIException(
+                AIException.ErrorCodes.UnknownError,
+                $"Something went wrong: {e.Message}", e);
+        }
+    }
+
+    /// <summary>
+    /// Run the HTTP request to generate a list of images
+    /// </summary>
+    /// <param name="url">URL for the chat request API</param>
+    /// <param name="requestBody">Request payload</param>
+    /// <param name="cancellationToken">Cancellation token</param>
+    /// <returns>List of images serialized in base64</returns>
+    /// <exception cref="AIException">AIException thrown during the request.</exception>
+    protected async Task<IList<string>> ExecuteImageBase64GenerationRequestAsync(
+        string url,
+        string requestBody,
+        CancellationToken cancellationToken = default)
+    {
+        try
+        {
+            var result = await this.ExecutePostRequestAsync<ImageUrlGenerationResponse>(url, requestBody, cancellationToken);
```

Comment:
We can rename `ImageUrlGenerationResponse` to `ImageGenerationResponse`, since it can contain URL response as well as base64 response.